allure-junit-adaptor/src/main/java/ru/yandex/qatools/allure/junit/AllureRunListener.java

```diff
@@ -107,7 +107,13 @@ public String generateSuiteUid(String suiteName) {
     public String getSuiteUid(Description description) {
         String suiteName = description.getClassName();
         if (!getSuites().containsKey(suiteName)) {
-            Description suiteDescription = Description.createSuiteDescription(description.getTestClass());
+            Class testClass = description.getTestClass();
+            Description suiteDescription;
+            if (testClass != null) {
+                suiteDescription = Description.createSuiteDescription(testClass);
+            } else {
+                suiteDescription = Description.createSuiteDescription(description.getClassName());
```

Comment:
In that case all the test without real class name will be in the same group in `xunit` tab, isn't?